backend/lib/geocoder.py

```diff
@@ -65,25 +62,37 @@ def geocode_batch(self, addresses):
                 for address in addresses
                 if results.get(address) and results.get(address).coords
             ]
-
-        except Exception:
+        if not results and len(addresses) > 0:
             logger.warning('Error in batch geocoding - switching to single geocoding.')
-
             with multiprocessing.dummy.Pool(processes=MAX_THREADS) as executor:
                 geocoded_addresses = executor.map(self.geocode, addresses)
 
-            return [
-                geocoded_address for geocoded_address
-                in geocoded_addresses if geocoded_address
-            ]
+        return [
+            geocoded_address for geocoded_address
+            in geocoded_addresses if geocoded_address
+        ]
 
-    @retry(
-        retry_on_exception=_retry_on_geocodio_errors,
-        wait_fixed=WAIT_FIXED_MILLISECONDS,
-        stop_max_attempt_number=STOP_MAX_ATTEMPT_NUMBER)
     def _safe_geocode(self, addresses):
-        # TODO: Handle more than 10,000 addresses at once.
-        return self.client.geocode(addresses)
+        if isinstance(addresses, (list, set)):
+            if len(addresses) > 1000:
+                raise GeocodioAuthError('Too many addresses. Switch to single geocoding.')
```

Comment:
should we really be raising a geocodio auth error? seems bad to me

Reply:
Simple GeocodioError instead?